aten/src/ATen/native/TensorFactories.cpp

```diff
@@ -514,6 +514,137 @@ Tensor& range_out(Tensor& result, Scalar start, Scalar end, Scalar step) {
   return at::_th_range_out(result, start, end, step);
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ triangle ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Tensor tril_indices(
+    int64_t row, int64_t col, int64_t offset, const TensorOptions& options) {
+  AT_CHECK(row >= 0, "row must be non-negative, got", row);
+  AT_CHECK(col >= 0, "col must be non-negative, got", col);
```

Comment:
Numpy supports 0 col and/or 0 row (try it out in a python interpreter!). Let's do the same

Reply:
Hi @zou3519, is this error below an expected behavior or a bug?

```python
>>> import torch
>>> torch.ones(0, 3)
tensor([], size=(0, 3))
>>> torch.ones(0, 3).tril(0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: invalid argument 1: expected a matrix at ../aten/src/TH/generic/THTensorMoreMath.cpp:1255
>>> torch.ones(1, 3).tril(0)
tensor([[1., 0., 0.]])
```

Reply:
Bug, looks like we don't support 0-sized dims in `torch.tril`. Please file a bug for it in our issue tracker.

Reply:
A good rule of thumb for operators like this are "does numpy support it?" In this case, numpy did support tril on `np.zeros([0, 3])`

Reply:
I see, thanks! I will file an issue for it.

Reply:
Issue added #15016